client/app/dbaas/logs/detail/streams/alerts/home/alerts-home.controller.js

```diff
@@ -1,9 +1,112 @@
-class LogsStreamsAlertsHomeCtrl {
-    constructor ($stateParams) {
+class AlertsHomeCtrl {
+    constructor ($state, $stateParams, $translate, CloudMessage, ControllerHelper, LogsStreamsService, StreamsAlertsConstant, StreamsAlertsService) {
+        this.$state = $state;
         this.$stateParams = $stateParams;
+        this.$translate = $translate;
+        this.CloudMessage = CloudMessage;
+        this.ControllerHelper = ControllerHelper;
+        this.LogsStreamsService = LogsStreamsService;
+        this.StreamsAlertsConstant = StreamsAlertsConstant;
+        this.StreamsAlertsService = StreamsAlertsService;
+
         this.serviceName = this.$stateParams.serviceName;
-        this.stream = this.$stateParams.streamId;
+        this.streamId = this.$stateParams.streamId;
+        this._initLoaders();
+    }
+
+    $onInit () {
+        this.runLoaders();
+    }
+
+    /**
+     * Runs all the loaded to fetch data from APIs
+     *
+     * @memberof AlertsHomeCtrl
+     */
+    runLoaders () {
+        this.alertIds.load();
+        this.stream.load();
+    }
+
+    /**
+     * initializes the alertsIDs and current stream
+     *
+     * @memberof AlertsHomeCtrl
+     */
+    _initLoaders () {
+        this.alertIds = this.ControllerHelper.request.getArrayLoader({
+            loaderFunction: () => this.StreamsAlertsService.getAlertIds(this.serviceName, this.streamId)
+        });
+        this.stream = this.ControllerHelper.request.getHashLoader({
+            loaderFunction: () => this.LogsStreamsService.getStream(this.serviceName, this.streamId)
+        });
+    }
+
+    /**
+     * Loads a number of alerts specified by the pageSize, starting from the specified offset
+     *
+     * @param {any} offset
+     * @param {any} pageSize
+     * @returns promise which will be resolve to the loaded alerts data
+     * @memberof AlertsHomeCtrl
+     */
+    loadAlerts ({ offset, pageSize }) {
+        return this.StreamsAlertsService.getAlerts(
```

Comment:
you can use a loader and set the meta property in the then

this.load().then()

Reply:
It did not strike me that I can use the ControllerHelper for some reason. I will make it a point to remember that. Thanks a lot for bringing this to attention :)